src/main/java/com/wavefront/opentracing/WavefrontSpanBuilder.java

```diff
@@ -131,24 +131,32 @@ public Span start() {
     if (globalTags != null && !globalTags.isEmpty()) {
       tags.addAll(globalTags);
     }
-    WavefrontSpanContext context = createSpanContext();
-    return new WavefrontSpan(tracer, operationName, context, startTimeMicros,
-        startTimeNanos, parents, follows, tags);
+    WavefrontSpanContext ctx = createSpanContext();
+    if (!ctx.isSampled()) {
+      // this indicates a root span and that no decision has been inherited from a parent span.
+      // perform head based sampling as no sampling decision has been obtained for this span yet.
+      boolean decision = tracer.sample(operationName, ctx.getTraceId().getLeastSignificantBits(), 0);
+      ctx = ctx.withSamplingDecision(decision);
+    }
+    return new WavefrontSpan(tracer, operationName, ctx, startTimeMicros, startTimeNanos, parents,
+        follows, tags);
   }
 
   private WavefrontSpanContext createSpanContext() {
-    UUID traceId = traceAncestry();
     UUID spanId = UUID.randomUUID();
-    return new WavefrontSpanContext(traceId, spanId);
+    WavefrontSpanContext traceCtx = traceAncestry();
+    UUID traceId = traceCtx == null ? UUID.randomUUID() : traceCtx.getTraceId();
```

Comment:
Usually try to add paranthesis for easier readability
see below
UUID traceId = (traceCtx == null) ? UUID.randomUUID() : traceCtx.getTraceId();